fix(navigation): resolve graph display issues when navigating between situational pages

anirudhlakhotia opened the pull request:
- Add proper state reset between page navigations
- Force component re-rendering with unique keys
- Improve component lifecycle handling in Results component
- Add navigation breadcrumbs and refresh buttons for better UX
- Enhance error handling and loading states during transitions

anirudhlakhotia committed:
fix(navigation): resolve graph display issues when navigating between situational pages

- Add proper state reset between page navigations
- Force component re-rendering with unique keys
- Improve component lifecycle handling in Results component
- Add navigation breadcrumbs and refresh buttons for better UX
- Enhance error handling and loading states during transitions

anirudhlakhotia committed:
Merge branch 'main' into new-fix-routing-branch

programmatix commented on vue/src/components/Results.vue:
```diff
@@ -126,19 +126,15 @@ export default {
-      this.fetchQuery(this.input)
-    }
+    console.info("Results component mounted");
+    this.forceRerender();
```
I'll merge anyway to keep us moving - but it feels like this and the unmounted() would ideally be superfluous.  There must be a way of getting Vue to create a fresh clean Results object, e.g. React key-style.

programmatix commented on vue/src/components/Results.vue:
```diff
@@ -216,14 +217,53 @@ export default {
   },
-  props: ['input', 'single']
+  props: ['input', 'single'],
+  watch: {
```
Ditto here.  We're adding a lot of logic to a single component, and it's unclear if this pattern needs to be applied to all components.  Ideally we would find a cleaner and more generic solution.

programmatix reviewed (approved):
A somewhat reluctant merge.  Surely Vue can't require all this.

programmatix commented:
Will merge once conflicts are solved.

anirudhlakhotia committed:
fix(router): stabilize situational routing and state propagation

anirudhlakhotia committed:
fix(results): prevent infinite rerender loops on /results pages

- Memoize computed query objects to avoid prop identity churn that retriggers fetches:
  - src/components/Shared.vue (kvInserts, kvReplaces, kvGets, kvGetsHorizontalScaling)
  - src/components/JavaClassic.vue (kvGetsHorizontalScalingAsync, kvGetsBlocking, kvGetsReactive)
  - src/components/Metrics.vue (processCpu, memHeapUsedMB, memRssUsedMB, threadCount)
- Caches now update only when relevant deps change (excludeSnapshots, language, excludeGerrit)
- Stabilizes props passed to Results.vue so its input watcher doesn't loop fetches
- Vite: set HMR clientPort to 8080 for local dev unless HTTPS to prevent reload loops

Impact: fixes infinite reloads on /results/java and /results/scala; reduces unnecessary API calls; improves UI stability.

anirudhlakhotia committed:
Merge branch 'couchbaselabs:main' into new-fix-routing-branch
